imcache_test.go

```diff
@@ -941,6 +942,53 @@ func TestImcache_GetAll_SlidingExpiration(t *testing.T) {
 	}
 }
 
+//GetAndRemove Tests Init
+func TestImcache_GetAndRemove(t *testing.T) {
+	for _, cache := range caches {
+		t.Run(cache.name, func(t *testing.T) {
+			c := cache.create()
+			c.Set("foo", "foo", WithNoExpiration())
+			c.Set("foobar", "foobar", WithExpiration(time.Hour))
+
+			//get foobar item and delete it
+			c.GetAndRemove("foobar")
+
+			got := c.GetAll()
+			want := map[string]string{
+				"foo": "foo",
+			}
+			if !reflect.DeepEqual(got, want) {
+				t.Errorf("imcache.GetAndRemove() = %v, want %v", got, want)
+			}
+		})
+	}
+}
+func TestImcache_GetAndRemove_Whit_Expiration(t *testing.T) {
```

Comment:
We don't really need a separate test, let's introduce something similar to what is already done for the `Remove` method (e.g. expired test case can be easily tested in a table-driven manner), see https://github.com/erni27/imcache/blob/master/imcache_test.go#L810.